wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/button/Trigger.java

```diff
@@ -139,23 +141,60 @@ public void run() {
    */
   public Trigger whileTrue(Command command) {
     requireNonNullParam(command, "command", "whileTrue");
-    m_loop.bind(
-        new Runnable() {
-          private boolean m_pressedLast = m_condition.getAsBoolean();
-
-          @Override
-          public void run() {
-            boolean pressed = m_condition.getAsBoolean();
+    return onTrue(new WrapperCommand(command) {
+      @Override
+      public boolean isFinished() {
+          return !m_condition.getAsBoolean() || super.isFinished();
+      }
+    });
+  }
```

Comment:
This does not behave the same. Previously when the trigger becomes false, the command would be cancelled. This makes it so the command appears to be finished to the scheduler, so `end(false)` is called.